### PR TITLE
OAK-6772: Convert oak-solr-core to OSGi R7 annotations

### PR DESCRIPTION
**Pull request by mbaedke**

### CODE REVIEW

File: oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/OakSolrConfigurationProviderService.java [hunk 1]
```diff
@@ -135,25 +176,25 @@ public class OakSolrConfigurationProviderService implements OakSolrConfiguration
...
-        useForPropertyRestrictions = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(PROPERTY_RESTRICTIONS)));
-        useForPrimaryTypes = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(PRIMARY_TYPES_RESTRICTIONS)));
-        typeMappings = PropertiesUtil.toStringArray(componentContext.getProperties().get(TYPE_MAPPINGS));
-        ignoredProperties = PropertiesUtil.toStringArray(componentContext.getProperties().get(IGNORED_PROPERTIES));
-        usedProperties = PropertiesUtil.toStringArray(componentContext.getProperties().get(USED_PROPERTIES));
-        propertyMappings = PropertiesUtil.toStringArray(componentContext.getProperties().get(PROPERTY_MAPPINGS));
-        collapseJcrContentNodes = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(COLLAPSE_JCR_CONTENT_NODES)));
-        collapseJcrContentParents = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(COLLAPSE_JCR_CONTENT_PARENTS)));
+    protected void activate(Configuration configuration) {
+        pathChildrenFieldName = configuration.path_child_field();
```
Comment (kwin) [inline oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/OakSolrConfigurationProviderService.java:180]: I would remove this indirection, just store the object

File: oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/RemoteSolrServerConfigurationProvider.java [hunk 2]
```diff
@@ -16,50 +16,80 @@
...
-    private static final String SOLR_CONF_DIR = "solr.conf.dir";
+        @AttributeDefinition(
+                name = "Solr configuration directory"
+        )
+        String solr_conf_dir() default SolrServerConfigurationDefaults.CONFIGURATION_DIRECTORY;
+
+        String name() default "remote";
+    }
 
     private String solrHttpUrl;
```
Comment (kwin) [inline oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/RemoteSolrServerConfigurationProvider.java:94]: I would remove all fields except for `remoteSolrServerConfiguration`

Comment (mbaedke) [reply oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/RemoteSolrServerConfigurationProvider.java:94]: That is also not in the scope of OAK-6772, is it?

File: oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/OakSolrConfigurationProviderService.java (hunk 1 above)
Comment (mbaedke) [reply oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/OakSolrConfigurationProviderService.java:180]: Imho hat would need a new Jira issue.

**Review by reschke — APPROVED** [review]
I agree that changes other than updating the annotations should have a separate issue and PR.

That said, I don't have suffient OSGi knowledge to review this. My recommendation would be to merge this if this passes tests, in particular OSGi it.

**Review by thomasmueller — APPROVED** [review]

**Review by fabriziofortino — APPROVED** [review]

**Review by rishabhdaim — APPROVED** [review]

**Pull request merged**